File: algorithms/go/sorting/quicksort.go
package main

import "fmt"

// Partition function for QuickSort
func partition(arr []int, low, high int) ([]int, int) {
	pivot := arr[high]
	i := low
	for j := low; j < high; j++ {
		if arr[j] < pivot {
			arr[i], arr[j] = arr[j], arr[i]
			i++
		}
	}
	arr[i], arr[high] = arr[high], arr[i]
	return arr, i
}

// QuickSort function
func quickSort(arr []int, low, high int) []int {
	if low < high {
		var p int
		arr, p = partition(arr, low, high)
		arr = quickSort(arr, low, p-1)
		arr = quickSort(arr, p+1, high)
	}
	return arr
}

func main() {
	arr := []int{10, 7, 8, 9, 1, 5}
	fmt.Println("Original array:", arr)
	
	arr = quickSort(arr, 0, len(arr)-1)
	
	fmt.Println("Sorted array:", arr)
}

File: algorithms/csharp/sorting/QuickSort.cs
using System;

namespace Algorithms.CSharp.Sorting
{
    class QuickSort
    {
        static int Partition(int[] arr, int low, int high)
        {
            int pivot = arr[high];
            int i = (low - 1);

            for (int j = low; j < high; j++)
            {
                if (arr[j] < pivot)
                {
                    i++;
                    int temp = arr[i];
                    arr[i] = arr[j];
                    arr[j] = temp;
                }
            }
            int temp1 = arr[i + 1];
            arr[i + 1] = arr[high];
            arr[high] = temp1;

            return i + 1;
        }

        static void Sort(int[] arr, int low, int high)
        {
            if (low < high)
            {
                int pi = Partition(arr, low, high);

                Sort(arr, low, pi - 1);
                Sort(arr, pi + 1, high);
            }
        }

        static void Main(string[] args)
        {
            int[] arr = { 10, 7, 8, 9, 1, 5 };
            Console.WriteLine("Original array: " + string.Join(", ", arr));

            Sort(arr, 0, arr.Length - 1);

            Console.WriteLine("Sorted array: " + string.Join(", ", arr));
        }
    }
}
